include/utils/math/numerical/numerical_integration.h

```diff
@@ -0,0 +1,234 @@
+#pragma once
+
+/**
+ * This header provides a variety of methods for solving ODEs depending on the
+ * needs of the system. First, second, and fourth order methods are
+ * provided, with first order being the fastest and least accurate, and fourth
+ * order being the slowest but most accurate.
+ * 
+ * Each method also provides a function for solving time-invariant ODEs:
+ * dx/dt = f(x, u)
+ * dx/dt = f(x)
+ * and time-variant ODEs:
+ * dy/dt = f(t, y)
+ * 
+ * Each function requires a function input, this can, and often is vector valued.
+ */
+
+
+
+/**
+ * Performs first order numerical integration of dx/dt = f(x, u) using Euler's
+ * method.
+ *
+ *   0|
+ * ---|---
+ *    |  1
+ *
+ * @param f The function to integrate, with two arguments x and u.
+ * @param x The initial value of x.
+ * @param u The input value u held constant over the integration period.
+ * @param dt The time over which to integrate.
+ */
+template <typename F, typename T, typename U>
+T euler(F &&f, T x, U u, double dt) {
+  double h = dt;
```

Comment:
Whats the rational for switching between `dt` and `h`? why not just name the argument `h` (or do math with `dt`)